contrib/test.bzl

```diff
@@ -0,0 +1,144 @@
+            remote = "https://github.com/GoogleCloudPlatform/container-structure-test.git",
+            tag = STRUCTURE_TEST_RELEASE,
+        )
+
+def _impl(ctx):
+    config_str = ' '.join(['$(pwd)/' + c.short_path for c in ctx.files.configs])
+
+    if ctx.attr.driver == 'tar':
+        # no need to load if we're using raw tar
+        load_statement = ''
+        image_name = '$(pwd)/' + ctx.file.image_tar.short_path
+
+    else:
+        # Since we're always bundling/renaming the image in the macro, this is valid.
+        load_statement = 'docker load -i %s' % ctx.file.image_tar.short_path
+        image_name = ctx.attr.image_name
+
+    # Generate a shell script to execute structure_tests with the correct flags.
+    ctx.actions.expand_template(
+        template = ctx.file._structure_test_tpl,
+        output = ctx.outputs.executable,
+        substitutions = {
+          "%{load_statement}": load_statement,
+          "%{configs}": config_str,
+          "%{workspace_name}": ctx.workspace_name,
+          "%{test_executable}": ctx.executable._structure_test.short_path,
+          "%{image}": image_name,
+          "%{driver}": ctx.attr.driver,
+        },
+        is_executable = True
+    )
+
+    return struct(runfiles = ctx.runfiles(files =
+            [
+                ctx.executable._structure_test,
+                ctx.executable.image_tar,
+                ctx.file.image_tar
+            ] +
+            ctx.attr.image_tar.files.to_list() +
+            ctx.attr.image_tar.data_runfiles.files.to_list() +
+            ctx.files.configs,
+        ),
+    )
+
+_container_test = rule(
+    attrs = {
+        "image_tar": attr.label(
+            executable = True,
+            allow_files = True,
+            mandatory = True,
+            single_file = True,
+            cfg = "target",
+        ),
+        "image_name": attr.string(
+            mandatory = True,
+        ),
+        "configs": attr.label_list(
+            mandatory = True,
+            allow_files = True,
+        ),
+        "verbose": attr.bool(
+            default = False,
+            mandatory = False,
+        ),
+        "driver": attr.string(
+            default = "docker",
+            doc = "Driver to use when running structure tests",
+            mandatory = False,
+            values = [
+                "docker",
+                "tar",
+            ],
+        ),
+        "_structure_test": attr.label(
+            default = Label("@structure_test//:go_default_test"),
+            cfg = "target",
+            executable = True,
+            allow_files = True,
+        ),
+        "_structure_test_tpl": attr.label(
+            default = Label("//contrib:structure-test.sh.tpl"),
+            allow_files = True,
+            single_file = True,
+        ),
+    },
+    executable = True,
+    test = True,
+    implementation = _impl,
+)
+
+def container_test(name, image, configs, driver=None, verbose=None, tags=[]):
```

Comment:
Can you set default value for driver ? Set one from the allowed list mentioned in the rule.